cppcache/src/TcrMessage.cpp

```diff
@@ -2090,17 +2090,16 @@ TcrMessageReply::TcrMessageReply(bool decodeAll,
 
 TcrMessagePing::TcrMessagePing(DataOutput* dataOutput, bool decodeAll) {
   m_msgType = TcrMessage::PING;
+  // -1 is NOTX constant server-side, PING should *never* be part of a
+  // transaction
+  m_txId = -1;
   m_decodeAll = decodeAll;
   m_request.reset(dataOutput);
   m_request->writeInt(m_msgType);
   m_request->writeInt(static_cast<int32_t>(
       0));  // 17 is fixed message len ...  PING only has a header.
   m_request->writeInt(static_cast<int32_t>(0));  // Number of parts.
-  // int32_t txId = TcrMessage::m_transactionId++;
-  // Setting the txId to 0 for all ping message as it is not being used on the
-  // SERVER side or the
-  // client side.
-  m_request->writeInt(static_cast<int32_t>(0));
+  m_request->writeInt(m_txId);
```

Comment:
Look at `TcrMessage::writeHeader` and see that almost all the messages are using this method to do some of the message bits in here. Perhaps it makes sense to normalize this message to match.